plone/resource/directory.py

```diff
@@ -265,23 +263,21 @@ def isFile(self, path):
         return os.path.isfile(self._resolveSubpath(path))
 
     def exportZip(self, out):
-        zf = zipfile.ZipFile(out, 'w')
-
-        toStrip = len(self.directory.replace(os.path.sep, '/')) + 1
+        with zipfile.ZipFile(out, 'w') as zf:
+            toStrip = len(self.directory.replace(os.path.sep, '/')) + 1
 
-        for (dirpath, dirnames, filenames) in os.walk(self.directory):
-            subpath = dirpath.replace(os.path.sep, '/')[toStrip:].strip('/')
+            for (dirpath, dirnames, filenames) in os.walk(self.directory):
+                subpath = dirpath.replace(os.path.sep, '/')[toStrip:].strip('/')
 
-            for filename in filenames:
-                path = '/'.join([subpath, filename]).strip('/')
+                for filename in filenames:
+                    path = '/'.join([subpath, filename]).strip('/')
 
-                if any(any(filter.match(n) for filter in FILTERS)
-                       for n in path.split('/')
-                       ):
-                    continue
+                    if any(any(filter.match(n) for filter in FILTERS)
+                           for n in path.split('/')
+                           ):
+                        continue
 
-                zf.writestr(
-                    '/'.join([self.__name__, path, ]),
-                    self.readFile(path)
-                )
-        zf.close()
+                    zf.writestr(
+                        '/'.join([self.__name__, path, ]),
```

Comment:
The zipfile docs say: "There is no official file name encoding for ZIP files. If you have unicode file names, you must convert them to byte strings in your desired encoding before passing them to write()."

Since directory listings are strings in Python 3 I think that means we need to encode the names for zipfile in Python 3 -- but I haven't actually tested.